source/renderer/app/components/wallet/settings/GenericPublicKeyField.js

```diff
@@ -0,0 +1,116 @@
+// @flow
+import React, { useCallback, useState } from 'react';
+import { observer } from 'mobx-react'; // 6.x or mobx-react-lite@1.4.0
+import SVGInline from 'react-svg-inline';
+import classnames from 'classnames';
+import { Button } from 'react-polymorph/lib/components/Button';
+import { Input } from 'react-polymorph/lib/components/Input';
+import { PopOver } from 'react-polymorph/lib/components/PopOver';
+import GenericPublicKeyFieldSkin from './GenericPublicKeyFieldSkin';
+import qrCodeImage from '../../../assets/images/qr-code.inline.svg';
+import revealKeyImage from '../../../assets/images/reveal-key.inline.svg';
+import hideKeyImage from '../../../assets/images/hide-key.inline.svg';
+import globalMessages from '../../../i18n/global-messages';
+import type { Locale } from '../../../../../common/types/locales.types';
+import { LOCALES } from '../../../../../common/types/locales.types';
+import styles from './GenericPublicKeyField.scss';
+
+type Props = {
+  publicKey: string,
+  locale: Locale,
+  onCopyPublicKey: Function,
+  onShowQRCode: Function,
+  onOpenWalletKeyDialog: Function,
+  t: Function,
```

Comment:
Please use context instead of props for `intl`. Check [this file](https://github.com/input-output-hk/daedalus/blob/develop/source/renderer/app/components/assets/AssetContent.js#L228).

Also, it's nice to be more descriptive in the prop names

Reply:
Since `t()` is wide used for translations (i18next) I just used it. But of course I will change it to intl()